### ISSUE TITLE
Dataset linked to a project is not visible in it

### ISSUE THREAD

**Issue by emeyke** [gbif/portal-feedback#2198]
E.g. dataset https://www.gbif.org/dataset/38c928d1-ae8b-46fb-a01a-08bd9cb00f73 is linked but not listed.

**Comment by dschigel** [comment]
The recent change of Hosted by preference to FinBIF in the Chronicles of Nature results in the invisibility of EarthCape in the example above, which is not very productive. This is linked to the request to feature Installations https://github.com/gbif/portal-feedback/issues/2197. Possibly relevant also for the Specify and Symbiota visibility (I am not sure).

**Comment by ManonGros** [comment]
Why would the change in hosting organisation make the dataset disappear from [the project page](https://www.gbif.org/project/sXNwwgP5NsFN5kxfdqnVq/chronicles-of-nature-of-russian-protected-areas-digitization-and-data-mobilization#datasets)?

**Comment by emeyke** [comment]
No, this is actually unrelated.
I think @dschigel may withdraw his comment :)

**Comment by dschigel** [comment]
Agreed comment is not related, but possibly still valid as standalone.

**Comment by ahahn-gbif** [comment]
I notice that the project title is missing in the non-linked datasets. That used to be an issue (both project id and title were required for correct mapping of datasets to the project pages), though I am not sure it should be any longer. Just an observation.

**Comment by emeyke** [comment]
Andrea, right, I assumed since it is linked by id, the title isn't needed in the metadata.
Did the link break because of that still?

**Comment by ahahn-gbif** [comment]
The project IDs look fine, and show up on the dataset page; it is not quite plausible why the link would work in one direction but not in the other, so I am a bit stumped what else might be causing this. The project title issue goes back more than a year, when the page layout (tabs) was still slightly different. If it is not too much hassle to add the titles, it is worth a try, but otherwise I wonder if @MortenHofft has an idea why some datasets do and others do not show up on the project's dataset tab.

**Comment by ahahn-gbif** [comment]
We are just dealing with a similar case, same problem (the dataset page contains the BID Project ID and links correctly to the project page, but the dataset is not listed under the project's "datasets" tab in one of the two cases):

- [does not show](https://api.gbif.org/v1/dataset/search?limit=500&projectId=BID-AF2020-122-NAC)
relates to 
[dataset](https://www.gbif.org/dataset/0201340c-c8df-433b-85b8-f2383a49cf20)
[project page](https://www.gbif.org/project/BID-AF2020-122-NAC/the-gabon-biodiversity-portal#datasets)
[GBIF-annotated EML](https://api.gbif.org/v1/dataset/0201340c-c8df-433b-85b8-f2383a49cf20/document)

- [does show](https://api.gbif.org/v1/dataset/search?limit=500&projectId=BID-CA2020-045-NAC)
relates to
[dataset](https://www.gbif.org/dataset/54f7af87-e0a3-4816-ae9f-ed77c45f2455)
[project page](https://www.gbif.org/project/BID-CA2020-045-NAC/data-mobilization-for-key-entomological-groups-across-caribbean-colombia#datasets)
[GBIF-annotated EML](https://api.gbif.org/v1/dataset/54f7af87-e0a3-4816-ae9f-ed77c45f2455/document)

I suspect that it has something to do with the resources' metadata, but we are stumped on what the difference would be between the two. Have we ever figured out what the cause was in the previous case?

**Comment by fmendezh** [comment]
I discussed this issue with @marcos-lg and we know now why the project Ids and other metadata information, extracted from EML files, are not being synchronised in the search index. It seems that Registry APIs only propagates changes made in the data base but it doesn't takes into consideration external changes.
This will be fixed by propagating dataset changes at the end of each crawling or  indexing job.

**Comment by ahahn-gbif** [comment]
Thanks, @fmendezh! I am not sure I fully understand yet. Indexing/crawling jobs on the dataset finished with "NOT MODIFIED" in the last months. The last actual run was in late February, with some manual follow-ups into April (https://registry.gbif.org/dataset/0201340c-c8df-433b-85b8-f2383a49cf20/ingestion-history). Do I understand correctly that this is then the reason that the search index did not capture the added project ID?

Since it is not possible to force-crawl a dataset (issuing a "crawl" from the ingestion page will report the same "not modified" and stop the process): does this mean that our search index will presently only catch up on the ProjectID when or if the actual content of the dataset next changes (which would then trigger a full indexing run)?

Is it that you are proposing to update our crawling and indexing behavior, so that external (dataset) changes are automatically included in future, regardless of the "modified" status of the dataset?

Many thanks!

**Comment by marcos-lg** [comment]
This was fixed in [this issue](https://github.com/gbif/registry/issues/425).

I also scheduled the reindexing all the datasets that have project ID. The example above was fixed. If you find more cases let me know (the reindexing will take some hours from now)
